icu4c/source/i18n/unicode/reldatefmt.h

```diff
@@ -696,11 +698,15 @@ class U_I18N_API RelativeDateTimeFormatter : public UObject {
     const SharedPluralRules *fPluralRules;
     UDateRelativeDateTimeFormatterStyle fStyle;
     UDisplayContext fContext;
+#if !UCONFIG_NO_BREAK_ITERATION
     const SharedBreakIterator *fOptBreakIterator;
```

Comment:
Please initialize the fOptBreakIterator with nullptr here in the class declaration and remove the constructor initializers. That will remove clutter because you won't need to ifdef each of those any more.

Reply:
but the type of  fOptBreakIterator is SharedBreakIterator  which is ifdef out

Reply:
I see. How about
```
#if UCONFIG_NO_BREAK_ITERATION
    std::nullptr_t fOptBreakIterator;
#else
    const SharedBreakIterator *fOptBreakIterator;
#endif // !UCONFIG_NO_BREAK_ITERATION
```
?
That will declutter *some* of the code, especially the initialization.
You should "only" need ifdefs where there is code with addRef(), removeRef(), and SharedObject::someThing().

Reply:
Used `std::nullptr_t`.